src/Forms/JoinSubtitles.cs

```diff
@@ -164,35 +155,22 @@ private void SortAndLoad()
 
                     if (format == null)
                     {
-                        for (int j = k; j < _fileNamesToJoin.Count; j++)
-                        {
-                            _fileNamesToJoin.RemoveAt(j);
-                        }
-                        MessageBox.Show("Unkown subtitle format: " + fileName);
+                        Revert(k, fileName);
                         break;
                     }
+
                     if (sub.Header != null)
                     {
                         header = sub.Header;
                     }
 
-                    if (lastFormat == null || lastFormat.FriendlyName == format.FriendlyName)
-                    {
-                        lastFormat = format;
-                    }
-                    else
-                    {
-                        lastFormat = new SubRip(); // default subtitle format
-                    }
+                    lastFormat = lastFormat == null || lastFormat.FriendlyName == format.FriendlyName ? format : new SubRip();
 
                     subtitles.Add(sub);
                 }
                 catch (Exception exception)
                 {
-                    for (int j = k; j < _fileNamesToJoin.Count; j++)
-                    {
-                        _fileNamesToJoin.RemoveAt(j);
-                    }
+                    Revert(k, exception.Message);
                     MessageBox.Show(exception.Message);

```

Comment:
Should this "MessageBox" be removed too?

Reply:
Done!